cmd/root.go

```diff
@@ -364,9 +364,17 @@ func newRunCommand(ctx context.Context, input *Input) func(*cobra.Command, []str
 		inputs := parseEnvs(input.inputs)
 		_ = readEnvs(input.Inputfile(), inputs)
 
-		log.Debugf("Loading secrets from %s", input.Secretfile())
 		secrets := newSecrets(input.secrets)
-		_ = readEnvs(input.Secretfile(), secrets)
+		log.Debugf("Loading secrets from %s", input.Secretfile())
+		secretsFromFile := make(map[string]string)
+		_ = readEnvs(input.Secretfile(), secretsFromFile)
+		for k, v := range secretsFromFile {
+			if _, ok := secrets[k]; ok {
+				log.Debugf("Ignoring secret %s from file in favor of secret from command-line", k)
+			} else {
+				secrets[k] = v
+			}
+		}
```

Comment:
Notably this is way more code than I would have liked, but the `readEnvs` function isn't composable in the other direction.

Upon further reflection, my guess is that `-e` has the same poor experience as `-s` which would argue in favor of reworking the function instead of changing the caller side -- but that'd require verifying that all callers are ok w/ the change, and I can't do anything like that at this time -- even this is giving me a headache.